deepseg.cc

```diff
@@ -298,15 +289,26 @@ void calc_mask(calcinfo_t &info, timinginfo_t &ti) {
 			out[n] = (val & 0xE0) | (out[n] >> 3);
 		}
 	}
-	ti.maskns=timestamp();
+	info.pti->maskns = diffnanosecs(timestamp(), t0);
 
 	// denoise
 	cv::Mat tmpbuf;
 	cv::dilate(info.ofinal,tmpbuf,info.element);
 	cv::erode(tmpbuf,info.ofinal,info.element);
 
 	// scale up into full-sized mask
-	cv::resize(info.ofinal,info.mroi,cv::Size(info.raw.rows/info.ratio,info.raw.rows));
+	pthread_mutex_lock(&info.lock_mask);
+	cv::resize(info.ofinal,info.mroi,cv::Size(raw.rows/info.ratio,raw.rows));
+	pthread_mutex_unlock(&info.lock_mask);
```

Comment:
Perhaps calculate a separate frame rate for mask generation (and display in stats)?

Reply:
Good Idea
Done